test/unit/modules/signatures.js

```diff
@@ -14,66 +14,226 @@
+					postSignatureStub.yields(null, { success: false });
 
-					it(
-						'should call callback with ApiError containing message = "Blockchain is loading"'
+					library.modules.signatures.shared.postSignature(
+						req.body,
+						(error, result) => {
+							expect(error).to.be.instanceof(ApiError);
+							done();
+						}
 					);
+				});
 
-					describe('when result.message = "Invalid signatures body"', () => {
-						it('should call callback with ApiError containing code = 400');
+				describe('when result.message = "Invalid signature body"', () => {
+					it('should call callback with ApiError containing code = 400', done => {
+						// Force library.modules.transport.shared.postSignature to fail with custom message
+						postSignatureStub.yields(null, {
+							success: false,
+							message: 'Invalid signature body',
+						});
+
+						library.modules.signatures.shared.postSignature(
+							req.body,
+							(error, result) => {
+								expect(error.code).to.equal(400);
+								done();
+							}
+						);
 					});
+				});
 
-					describe('when result.message != "Invalid signatures body"', () => {
-						it('should call callback with ApiError containing code = 500');
+				describe('when result.message != "Invalid signature body"', () => {
+					it('should call callback with ApiError containing code = 500', done => {
+						// Force library.modules.transport.shared.postSignature to fail with custom message
+						postSignatureStub.yields(null, {
+							success: false,
+							message: 'A different message',
+						});
+
+						library.modules.signatures.shared.postSignature(
+							req.body,
+							(error, result) => {
+								expect(error.code).to.equal(500);
+								done();
+							}
+						);
 					});
 				});
+			});
+
+			describe('when modules.transport.shared.postSignature succeeds with result', () => {
+				let postSignatureError;
+				let postSignatureResult;
 
-				describe('when modules.transport.shared.postSignatures succeeds with result', () => {
-					it('should call callback with error = null');
+				before(done => {
+					const postSignatureStub = sinonSandbox.stub(
+						library.modules.transport.shared,
+						'postSignature'
+					);
+					// Force library.modules.transport.shared.postSignature to succeed
+					postSignatureStub.yields(null, {
+						success: true,
+					});
 
-					it(
-						'should call callback with result containing status = "Signature Accepted"'
+					library.modules.signatures.shared.postSignature(
+						req.body,
+						(error, result) => {
+							postSignatureError = error;
+							postSignatureResult = result;
+							done();
+						}
 					);
 				});
+
+				it('should call callback with error = null', done => {
+					expect(postSignatureError).to.be.null;
+					done();
+				});
+
+				it('should call callback with result containing status = "Signature Accepted"', done => {
+					expect(postSignatureResult.status).to.equal('Signature Accepted');
+					done();
```

Comment:
Please apply my comments above to this file as well.

Reply:
Done